crd-generator/maven-plugin/src/test/java/io/fabric8/crd/generator/maven/plugin/ClasspathTypeTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crd.generator.maven.plugin;
+
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.model.Build;
+import org.apache.maven.project.MavenProject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledForJreRange;
+import org.junit.jupiter.api.condition.JRE;
+import org.mockito.Mockito;
+
+import java.util.Collections;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@EnabledForJreRange(min = JRE.JAVA_17)
```

Comment:
Why do we want to run these tests in Java 17 and above only? Is it possible to remove it?

Reply:
I get some mockito related errors and havn't figured out yet why this is only the case on a lower versions than 17. (see https://github.com/fabric8io/kubernetes-client/actions/runs/9052896736/job/24871094873#step:4:36770)
Any ideas? 

But in general I'm not really happy with both tests in the maven-plugin module. A lot stuff must be mocked, only to test some simple things...

I'm open to rewrite both tests enterily if someone can point me into the right direction. e.g. writing e2e tests with tool x. (The java-generator maven plugin doesn't include any own tests, so I cannot use it as template)

Reply:
Do you think you can use [maven invoker plugin](https://maven.apache.org/plugins/maven-invoker-plugin/) here? We use it for [JKube  integration tests](https://github.com/eclipse-jkube/jkube/blob/master/kubernetes-maven-plugin/it/pom.xml#L67) to verify generated YAML files.

Reply:
Added the first integration tests, is this the right direction?

Reply:
In my opinion, it looks okay, but we might want to get opinions from other team members. 

Maybe we can make the verification check more strict (it might help us find the differences in generated YAML). In Eclipse JKube we compare whole YAML manifests in `expected/` directory with generated output.

Reply:
It's up to you. If you think they're redundant feel free to remove them.

Reply:
All `@EnabledForJreRange(min = JRE.JAVA_17)` annotations are removed, no more java version dependent tests.
I think this conversion can be resolved.